### ISSUE TITLE
Globalization support?

### ISSUE THREAD

**Issue by mtayse** [phax/peppol-smp-server#75]
Hi,
I know there is some UI in your SMP server, for example to display the participants managed on a given SMP.  Can you comment on state, and/or plans, for G11n support?  I am most interested in languages available--I know of English, are there any others?   

I'm also interested, where applicable, of other G11n areas, such as unicode support, bidi, UI mirroring, date formats, and calendar systems; does the SMP support these, to the extent they are supported by the run-time environment?

Thank you

**Comment by phax** [comment]
Hi,
* Language is currently hard coded to English. Other languages are possible but currently not forseen. That also has impact on base libraries that provided commons texts (mostly German and English)
* Unicode support is present in all cases - everything is UTF-8 (stored data, user interface, REST API request and response)
* Bidi should not be an issue because of UTF-8 usage - except the layout maybe a bit broken, but nothing that can't be fixed.... Never used it though.
* UI mirroring - please elaborate, I don't know what you mean
* Date formats: on the UI the date formats depend on the user interface "display locale" - currently only English. But it works for all date formats supplied by the underlying JDK
* Different calendar systems: the usage of the Gregorian Calendar is imho necessary (this heavily depends on the usage of other Calendar systems in X.509 certificates - I have no clue).

Basically it's a  Java 8 application with general G11 in mind, but English only UI atm. The standardized parts (OASIS BDXR SMP v1 and PEPPOL SMP specification) are all language independent.

hth, Philip

**Comment by mtayse** [comment]
Thanks Philip.  Sorry I must've missed your comment in July.  

I think it's fine to be English-only (as long as the EU is ok with it, and they surely must be).  I mostly just needed to know, as the compliance people in my company will eventually ask why we're not translating our app.  Are the strings externalized?

FYI UI Mirroring is flipping the UI left for right, useful in languages that read right to left, like Arabic or Hebrew.  It's related to bidi, only more so....Here's a reasonably clear article with examples https://blogs.technet.microsoft.com/office_global_experience/2010/02/12/qa-whats-the-difference-between-ui-flipping-and-ui-mirroring/ .

**Comment by phax** [comment]
Well, mirroring is not really supported out of the box, but it should be doable with simple CSS tricks as shown in http://cssdeck.com/labs/mirror-effect :
```
body {
	-webkit-transform: matrix(-1, 0, 0, 1, 0, 0);
	/* Last 2 values in Mozilla requires a unit */
	-moz-transform: matrix(-1, 0, 0, 1, 0px, 0px);
	-ms-transform: matrix(-1, 0, 0, 1, 0, 0);
	-o-transform: matrix(-1, 0, 0, 1, 0, 0);
	transform: matrix(-1, 0, 0, 1, 0, 0);
}
```

**Comment by phax** [comment]
The strings of the generic base components support for externalization, whereas the application specific strings are not externalized.
